src/Command/GenerateObjectAclCommand.php

```diff
@@ -50,14 +51,33 @@ class GenerateObjectAclCommand extends QuestionableCommand
     private $aclObjectManipulators = [];
 
     /**
-     * @var RegistryInterface
+     * @var RegistryInterface|ManagerRegistry|null
      */
     private $registry;
 
-    public function __construct(Pool $pool, array $aclObjectManipulators, RegistryInterface $registry = null)
+    /**
+     * @param RegistryInterface|ManagerRegistry|null $registry
+     */
+    public function __construct(Pool $pool, array $aclObjectManipulators, $registry = null)
     {
         $this->pool = $pool;
         $this->aclObjectManipulators = $aclObjectManipulators;
+        if (null !== $registry && (!$registry instanceof RegistryInterface || !$registry instanceof ManagerRegistry)) {
+            if ($registry instanceof RegistryInterface) {
+                @trigger_error(sprintf(
+                    'Using the "%s" class is deprecated since, use "%s" instead.',
```

Comment:
```suggestion
                    'Passing an object that doesn\'t implement %s as argument 3 to %s() is deprecated since sonata-project/admin-bundle 3.x.',
```